File: verve patch/base/Applications/CSymphony_0_1/vbSamples/CodePropertySetValueReferenceExpression.vb
Option Strict Off
Option Explicit On

Imports System
Imports System.CodeDom
Imports System.CodeDom.Compiler
Imports System.Collections
Imports CSymphony.Lib

Namespace CSymphony.CSy
	Public Class CSyUnit
		 Inherits GraphUnit
		Overrides Public Sub Build()
			ns("MyNameSpace")

			declClass(TypePublic, "Type1")
			
			declField(MemPrivate, "System.Int32", "integerField")

			Dim intFieldRef As FieldRef = New FieldRef (my(), "integerField")

			declCtor(MemPublic)
			
			declProperty(MemPrivate, "System.Int32", "integerProperty")
			
			declGet()
			rtn(intFieldRef.Expr())

			declSet()
			enter (equate(intFieldRef.Expr(), propval()))
		End Sub
	End Class
End Namespace
			

File: verve patch/base/Applications/CSymphony_0_1/csSamples/CodePropertySetValueReferenceExpression.cs
using System;
using System.CodeDom;
using System.CodeDom.Compiler;
using System.Collections;
using CSymphony.Lib;

namespace CSymphony.CSy
{
	public class CSyUnit : GraphUnit
	{
		public override void Build()
		{
			ns("MyNameSpace");

			declClass(TypePublic, "Type1");
			FieldType intField = declField(MemPrivate, "System.Int32", "integerField");

			declCtor(MemPublic);
			
			declProperty(MemPrivate, "System.Int32", "integerProperty");
			
			declGet();
			rtn(intField);

			declSet();
			enter (equate(intField, propval()));

		}
	}
}
